apps/mull-ui/src/app/components/event-card/event-card.tsx

```diff
@@ -3,20 +3,29 @@ import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { ISerializedEvent } from '@mull/types';
 import React, { useState } from 'react';
 import { dummyProfilePictures } from '../../../constants'; // TODO query the participants profile pictures
+import { useJoinEventMutation, useLeaveEventMutation } from '../../../generated/graphql';
 import { formatDate } from '../../../utilities';
 import EventMembers from '../event-members/event-members';
 import './event-card.scss';
 
 export interface EventCardProps {
   event: Partial<ISerializedEvent>;
   style?: React.CSSProperties;
+  isJoined?: boolean;
   onClick?: (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void;
 }
 
-export const EventCard = ({ event, style = {}, onClick }: EventCardProps) => {
-  // TODO; set joined based on if current user is part of event
-  const [joined, setJoined] = useState<boolean>(false);
+export const EventCard = ({ event, style = {}, onClick, isJoined = false }: EventCardProps) => {
   const { day, month, time } = formatDate(new Date(event.startDate));
+
+  const [joined, setJoined] = useState<boolean>(isJoined);
+  const eventId = Number(event.id);
```

Comment:
Isn't `event.id` already a number? If it's not. Then the TypeScript type annotation of ISerializedEvent is wrong and needs to be tweaked

EDIT: just confirmed that it's a string, ISerializedEvent needs to be updated so that this is reflected.

Not gonna block over this. If you don't address this in this PR then please make a ticket for it

Reply:
I will create a ticket for it

Reply:
Addressed